<a href="https://colab.research.google.com/github/haardehola/HamoyeLearning/blob/main/Weekly_ML_Code_Club_001.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Part One: Load Data**

In [1]:
import numpy as np
import pandas as pd

In [2]:
cell_phones  = pd.read_csv('https://raw.githubusercontent.com/zalihat/cell_phone_dataset/master/cell_phones.csv')
cell_phones.head()    #checks the head of this dataframe

Country  year  cell_phones_total
0  Afghanistan  1960                0.0
1  Afghanistan  1961                0.0
2  Afghanistan  1962                0.0
3  Afghanistan  1963                0.0
4  Afghanistan  1964                0.0

In [3]:
population = pd.read_csv('https://raw.githubusercontent.com/zalihat/cell_phone_dataset/master/population1.csv')
population.head()     #checks the head of this dataframe

Country  year  population
0  Afghanistan  1800     3280000
1  Afghanistan  1801     3280000
2  Afghanistan  1802     3280000
3  Afghanistan  1803     3280000
4  Afghanistan  1804     3280000

In [4]:
geo_country = pd.read_csv('https://github.com/open-numbers/ddf--gapminder--systema_globalis/raw/master/ddf--entities--geo--country.csv')
geo_country.head()    #checks the head of this dataframe

country g77_and_oecd_countries  ... world_4region        world_6region
0    abkh                 others  ...        europe  europe_central_asia
1     abw                 others  ...      americas              america
2     afg                    g77  ...          asia           south_asia
3     ago                    g77  ...        africa   sub_saharan_africa
4     aia                 others  ...      americas              america

[5 rows x 21 columns]

# **Part Two: Join Data**

In [5]:
cell_population = cell_phones.merge(population, how='inner', on=['Country', 'year'])
cell_population.head()    #checks the head of this dataframe

Country  year  cell_phones_total  population
0  Afghanistan  1960                0.0     9000000
1  Afghanistan  1961                0.0     9170000
2  Afghanistan  1962                0.0     9350000
3  Afghanistan  1963                0.0     9540000
4  Afghanistan  1964                0.0     9740000

In [6]:
cell_population.shape     #checks the shape of the resulting dataframe

(11640, 4)

In [7]:
geo_country.rename(columns={'country':'geo', 'name':'Country'}, inplace=True)

In [8]:
selected_cols = geo_country[['geo', 'Country']]
cell_population_geo = cell_population.merge(selected_cols)
cell_population_geo.head()    #checks the head of this dataframe

Country  year  cell_phones_total  population  geo
0  Afghanistan  1960                0.0     9000000  afg
1  Afghanistan  1961                0.0     9170000  afg
2  Afghanistan  1962                0.0     9350000  afg
3  Afghanistan  1963                0.0     9540000  afg
4  Afghanistan  1964                0.0     9740000  afg

In [9]:
cell_population_geo.shape     #checks the shape of the resulting ddataframe

(11640, 5)

# **Part Three: Make Features**

In [11]:
cell_population_geo['cell_phones_per_person'] = cell_population_geo['cell_phones_total'] / cell_population_geo['population']

In [12]:
cell_population_geo[(cell_population_geo['year'] == 2017) & (cell_population_geo['Country'] == 'United States')]
#checks assertation

Country  year  ...  geo  cell_phones_per_person
11157  United States  2017  ...  usa                1.230769

[1 rows x 6 columns]

In [13]:
cell_population_geo['geo'] = cell_population_geo['geo'].str.upper()

# **Part Four: Process Data**

In [14]:
cell_population_geo.describe(include=[np.number])   #includes only numeric columns

year  cell_phones_total    population  cell_phones_per_person
count  11640.000000       1.164000e+04  1.164000e+04            11640.000000
mean    1989.500000       7.875722e+06  2.710307e+07                0.235536
std       17.318846       5.572496e+07  1.086187e+08                0.433303
min     1960.000000       0.000000e+00  4.380000e+03                0.000000
25%     1974.750000       0.000000e+00  1.117500e+06                0.000000
50%     1989.500000       0.000000e+00  5.100000e+06                0.000000
75%     2004.250000       5.615000e+05  1.590000e+07                0.233667
max     2019.000000       1.730000e+09  1.430000e+09                2.126068

In [15]:
cell_population_geo.describe(exclude=[np.number])   #includes non-numeric columns

Country    geo
count    11640  11640
unique     194    194
top       Chad    NLD
freq        60     60

In [16]:
cell_population_geo.describe(include='all')   #includes both numeric and non-numeric columns

Country          year  ...    geo  cell_phones_per_person
count    11640  11640.000000  ...  11640            11640.000000
unique     194           NaN  ...    194                     NaN
top       Chad           NaN  ...    NLD                     NaN
freq        60           NaN  ...     60                     NaN
mean       NaN   1989.500000  ...    NaN                0.235536
std        NaN     17.318846  ...    NaN                0.433303
min        NaN   1960.000000  ...    NaN                0.000000
25%        NaN   1974.750000  ...    NaN                0.000000
50%        NaN   1989.500000  ...    NaN                0.000000
75%        NaN   2004.250000  ...    NaN                0.233667
max        NaN   2019.000000  ...    NaN                2.126068

[11 rows x 6 columns]

Assertation: Year period ranges from 1960(min) to 2019(max) and there are 194 unique countries.

In [10]:
cell_population_geo[cell_population_geo['year'] == 2017].nlargest(5, ['cell_phones_total'])[['Country', 'cell_phones_total']].style.hide_index()